root/edit/details/AddWork.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * @flow
+ * Copyright (C) 2020 Anirudh Jain
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import EntityLink from '../../static/scripts/common/components/EntityLink';
+import {commaOnlyListText} from
+  '../../static/scripts/common/i18n/commaOnlyList';
+
+
+ type AddWorkEdit = {
+   ...EditT,
+   +display_data: {
+     +attributes?: {
+       [key: string]: $ReadOnlyArray<WorkAttributeT>,
+      },
+     +comment: string,
+     +iswc: string,
+     +language?: LanguageT,
+     +languages?: $ReadOnlyArray<string>,
+     +name: string,
+     +type: WorkTypeT | null,
+     +work: WorkT,
+   },
+ };
+
+
+const AddWork = ({edit}: {+edit: AddWorkEdit}) => {
+  const display = edit.display_data;
+  const attributes = display.attributes ?? {};
+  const type = display.type;
+  const language = display.language;
+  return (
+    <>
+      <table className="details">
+        <tr>
+          <th>{addColonText(l('Work'))}</th>
+          <td><EntityLink entity={display.work} /></td>
+        </tr>
+      </table>
+      <table className="details add-work">
+        <tr>
+          <th>{addColonText(l('Name'))}</th>
+          <td>{display.name}</td>
+        </tr>
+        {display.comment ? (
+          <tr>
+            <th>{addColonText(l('Disambiguation'))}</th>
+            <td>{display.comment}</td>
+          </tr>
+        ) : null}
+        {display.iswc ? (
+          <tr>
+            <th>{addColonText(l('ISWC'))}</th>
+            <td>{display.iswc}</td>
+          </tr>
+        ) : null}
+        {type ? (
+          <tr>
+            <th>{addColonText(l('Type'))}</th>
+            <td>{lp_attributes(type.name, 'work_type')}</td>
+          </tr>
+        ) : null}
+        {language ? (
+          <tr>
+            <th>{addColonText(l('Language'))}</th>
+            <td>{language.name}</td>
+          </tr>
+        ) : null}
+        {display.languages && display.languages.length ? (
+          <tr>
+            <th>{addColonText(l('Lyrics Languages'))}</th>
+            <td>
+              {commaOnlyListText(display.languages ? display.languages.map(
+                language => l_languages(language),
+              ) : [])}
+            </td>
+          </tr>
+        ) : null}
+        {Object.keys(attributes).sort().map((attributeType) => (
```

Comment:
Would also suggest using `attributeName` instead of `attributeType` here to match the type indexer name (and since we often use `fooType` when it's the actual type object, so helps disambiguate things slightly).